### ISSUE TITLE
The quick_actions plugin does not seem to support the Navigator

### ISSUE THREAD

**Issue by GroovinChip** [flutter/flutter#25348]
When using this package in conjunction with the Navigator, nothing happens. No crash, no logs, nothing. Here's a sample:
```dart
  @override
  void initState() {
    super.initState();
    final QuickActions quickActions = const QuickActions();
    quickActions.initialize((String shortcutType) {
      if (shortcutType == 'new_call') {
        print("shortcut");
        Navigator.pushNamed(context, '/AddNewCallScreen');
      }
    });

    quickActions.setShortcutItems(<ShortcutItem>[
      const ShortcutItem(
        type: 'new_call',
        localizedTitle: 'New Call',
        icon: "Plus",
      ),
    ]);
  }
```
the `print` works, but the Navigator is not activated.

Flutter Doctor results:
```
[√] Flutter (Channel beta, v1.0.0, on Microsoft Windows [Version 10.0.17134.407], locale en-US)
[√] Android toolchain - develop for Android devices (Android SDK 28.0.3)
[√] Android Studio (version 3.2)
[√] Android Studio (version 3.2)
[!] VS Code (version 1.29.1)
[√] Connected device (1 available)

! Doctor found issues in 1 category.
```

**Comment by zoechi** [comment]
Could you please create minimal runnable reproduction
as a single file so that we can just copy your code into `lib/main.dart` of a new project and run to reproduce.

**Comment by GroovinChip** [comment]
Done. You can view the application here:
https://github.com/GroovinChip/flutter-quick-actions-navigator-issue

**Comment by GroovinChip** [comment]
sorry I just noticed that you wanted it all in one file, would you like me to update the project? Or do you want to just clone and run it.

**Comment by zoechi** [comment]
@GroovinChip looks good to me as it is.

**Comment by GroovinChip** [comment]
@zoechi alright. what do you think the problem might be? I imagine something about the way dart code is executed in the background?

**Comment by GroovinChip** [comment]
Is there an update on this?

**Comment by bestK** [comment]
some problem  😩

**Comment by britannio** [comment]
https://api.flutter.dev/flutter/widgets/State/initState.html states that BuildContext.dependOnInheritedWidgetOfExactType cannot be used in `initState()` however it can be used inside `didChangeDepencencies()` so since the navigator relies on context, the navigator cannot be used inside `initState()`.

A suitable error message would be ideal.

**Comment by darshankawar** [comment]
Issue replicable on latest stable (1.20.2). No action takes place.

**Comment by android404base** [comment]
I reproduced the issue with the example made by GroovinChip  https://github.com/flutter/flutter/issues/25348#issuecomment-447335525 and it seems to be fixed in version 0.3.0. 

Could you verify that it is fixed?

**Comment by mvanbeusekom** [comment]
As mentioned by @android404base this issue seems to be resolved starting from version 0.3.0 of the quick_actions plugin. I can confirm that after upgrading the quick_actions plugin in the example project provided by @GroovinChip to version 0.3.0 the issue doesn't occur anymore and I will therefore close this issue. 

If you feel this is not correct please let me know and I would be happy to reopen the issue.

**Comment by GroovinChip** [comment]
I'll check this out later today and get back to you guys

**Comment by mvanbeusekom** [comment]
Maybe useful to know that I had to migrate the original example App to support AndroidX using the following steps:

1. Open the Android project using Android Studio (in Android Studio select "open existing project" and navigate to the "android" folder to open the project);
2. Update the `compileSdkVersion` to `28` (in the `android/app/build.gradle` file);
3. Select "Migrate to AndroidX..." from the "Refactor" menu. 

After that I was able to run the project on my RedMi Note 9 test device and the Android emulator.
